lib/cosmos/tools/tlm_viewer/tlm_viewer.rb

```diff
@@ -53,6 +53,10 @@ class TlmViewer < QtTool
     @@instance = nil
 
     def self.instance
+      unless @@instance
+        _, options = create_default_options()
+        TlmViewer.new(options)
```

Comment:
TlmViewer#initialize does a whole bunch of stuff you don't want to do / or waste time doing for just the --screen case.   A new option probably needs to be added before calling TlmViewer.new that only does the minimal stuff.

Reply:
I want a full instance of TlmViewer so the next guy who calls screen just works. That way the user can also use it and there's a connection back to the server. Otherwise you have this magic screen with a hidden TlmViewer backend? Then we have to figure out how to kill that.

Reply:
Yes - There is supposed to be a non-listening hidden TlmViewer backend.   That way you can call --screen lots of times and not open any sockets.   I agree that things like buttons that open other screens should work.  But there should not be a main TlmViewer screen chooser window, and it should not listen to the socket.

Reply:
@ryanatball I'm going to punt this back to you then. I'm not sure how this might dovetail with the --screen option we already have which looks like it's setting up a standalone backend .